### PR TITLE
Update node details under clusters

### PR DESCRIPTION
**Pull request by nnDarshan**
Node agent should update the node details under the
clusters in etcd. Node agent should check if it has
integration id under its tendrl_context, if available
then it should update the node_context for that node
at /clusters/<cluster-id>/nodes/<node-id>/NodeContext

tendrl-bug-id: Tendrl/node-agent#226
Signed-off-by: nnDarshan <darshan.n.2024@gmail.com>

### CODE REVIEW

**Comment by nnDarshan** [comment]
@r0h4n @nthomas-redhat @shtripat Please review

File: tendrl/node_agent/objects/cluster_node_context/__init__.py~ [hunk 1]
```diff
@@ -0,0 +1,71 @@
+import logging
```
Comment (shtripat) [inline tendrl/node_agent/objects/cluster_node_context/__init__.py~]: Is this file also required?

File: tendrl/node_agent/node_sync/__init__.py [hunk 2]
```diff
@@ -62,6 +62,17 @@ def _run(self):
                 tendrl_ns.node_agent.objects.NodeContext(tags=tags).save()
                 gevent.sleep(interval)
 
+                if tendrl_ns.tendrl_context.integration_id:
```
Comment (r0h4n) [inline tendrl/node_agent/node_sync/__init__.py]: You had to also check if the integration_id actually exists at /clusters/$integration_id, please do that

File: tendrl/node_agent/objects/cluster_node_context/__init__.py~ (hunk 1 above)
Comment (r0h4n) [inline tendrl/node_agent/objects/cluster_node_context/__init__.py~]: This file required?

Comment (nnDarshan) [inline tendrl/node_agent/objects/cluster_node_context/__init__.py~]: will remove

Comment (nnDarshan) [inline tendrl/node_agent/objects/cluster_node_context/__init__.py~]: will remove

File: tendrl/node_agent/node_sync/__init__.py (hunk 2 above)
Comment (nnDarshan) [inline tendrl/node_agent/node_sync/__init__.py]: Done

File: tendrl/node_agent/node_sync/__init__.py [hunk 3]
```diff
@@ -62,6 +62,28 @@ def _run(self):
...
+                    try:
+                        tendrl_ns.etcd_orm.client.read(
+                            "/clusters/%s" % (
+                                tendrl_ns.tendrl_context.integration_id
+                            )
+                        )
+                    except etcd.EtcdKeyNotFound:
+                        pass
```
Comment (r0h4n) [inline tendrl/node_agent/node_sync/__init__.py]: Need a log.error here saying that local tendrl context not found in central store

Comment (nnDarshan) [inline tendrl/node_agent/node_sync/__init__.py]: Done

**Pull request merged**